New styling and icons. Bug fix. DQA

Camwyn opened the pull request:
[EXT-241]

[EXT-241]: https://moderntribe.atlassian.net/browse/EXT-241

This PR adds:

- A new view icon.
- A few small style thingies from the last DQA.

and fixes:
- The constants containing "COMPACT" - as it's now called "SUMMARY" TYVM.
- A bug where a multiday event will not show on a day (other than it's start date) where there are no other events.
  - 📷 http://p.tri.be/xd9mSA
- This also adds auto-enabling the view on plugin activation. 
  - 🎥 http://p.tri.be/oo21Ux

**DQA update:** 
:camera: http://p.tri.be/D6AZCf
:camera: http://p.tri.be/LRBxsc

Commit:
New styling and icons.

Commit:
Test

Commit:
Merge branch 'feature/styles' of github.com:mt-support/tribe-ext-summary-view into feature/styles

Commit:
Small style tweaks, quiet a notice about a param

Commit:
Add missing navigation icons.

Commit:
fix icon spacing.

Commit:
DQA

Camwyn commented:
Marking this "ready for review since I did change a couple small php things. 
Can always create a DQA PR as needed.

Camwyn commented on src/Tribe/Plugin.php:
```diff
@@ -92,7 +92,7 @@ public function register() {
 		$this->container->singleton( 'extension.summary_view', $this );
 		$this->container->singleton( 'extension.summary_view.plugin', $this );
-		$this->container->register( PUE::class );
+		//$this->container->register( PUE::class );
```
Not sure this should be in here? licensing?

Camwyn commented on src/Tribe/Views/Summary_View.php:
```diff
@@ -191,7 +191,7 @@ protected function get_events_that_start_and_end_around( $first_date, $last_date
 	 * @param array $exclude_ids Event IDs in result set.
 	 *
-	 * @return array
+	 * @return \WP_Post
```
This was throwing param mismatch warnings

Camwyn commented on src/views/v2/components/icons/summary.php:
```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * View: Summary Icon
```
MCR is supposed to provide a tweaked version of this.

Camwyn commented on src/views/v2/summary/event/date.php:
```diff
@@ -23,21 +23,22 @@
 			<span class="tribe-event-date-start"><?php echo esc_html( $event->summary_view->start_time ); ?></span> -
 			<span class="tribe-event-date-end"><?php echo esc_html( $event->summary_view->end_time ); ?></span>
-		<?php elseif ( $event->summary_view->start_date === $formatted_group_date ) : ?>
+		<?php elseif ( ! empty( $event->summary_view ) && $event->summary_view->start_date === $formatted_group_date ) : ?>
```
As discussed - this sometimes throws a warning, so we now check `empty()` before checking the value.

Commit:
Fix infinite loop?

Commit:
New icon.

Camwyn committed:
Updates and bugfix.

This PR fixes:
A bug where a multiday event will not show on a day (other than it's start date) where there are no other events. :camera: http://p.tri.be/xd9mSA
The constants containing "COMPACT" - as it's now called "SUMMARY" TYVM.
A few small style thingies from the last DQA.

This also adds auto-enabling the view on plugin activation. :movie_camera: http://p.tri.be/oo21Ux

borkweb commented on plugin.php:
```diff
@@ -72,3 +72,17 @@ function tribe_extension_summary_view() {
+ */
+function tribe_extension_summary_view_activation() {
+	$enabled_views   = tribe_get_option('tribeEnableViews');
+	$enabled_views[] = 'summary';
```
```suggestion
	if ( in_array( 'summary', $enabled_views ) ) {
		return;
	}
	
	$enabled_views[] = 'summary';
```

Camwyn replied:
corrected (above marked line)

borkweb commented on src/Tribe/Plugin.php:
```diff
@@ -144,4 +138,11 @@ protected function register_plugin_dependencies() {
 		$this->container->singleton( 'extension.summary_view', $plugin_register );
 	}
+
+	protected function plugin_activated() {
```
Add a docblock here

Camwyn replied:
Whups - removing this, it's unused.

borkweb reviewed (changes requested):
This looks pretty great! Just a couple of tweaks.

Camwyn committed:
Remove unused function.
Add check for if view is already set/enabled.

Camwyn committed:
Remove icons from date - keep just the ones on the title.
Style tweaks - DQA.

:camera: http://p.tri.be/D6AZCf
:camera: http://p.tri.be/LRBxsc

borkweb reviewed (approved):
:+1:
